kernel/Makefile

```diff
@@ -21,10 +21,9 @@ lib-y += $(strip \
 	queue.o \
 	stack.o \
 	mem_slab.o \
-	mem_pool.o \
+	mempool.o \
```

Comment:
Simply for clarity of review.  With the exception of k_malloc()/k_free (which are simple wrappers) all the code is new.  Easier to read that in a single chunk than a bunch of synthesized diffs.